dashboards/rbac/rbac_explorer.sp

```diff
@@ -0,0 +1,397 @@
+dashboard "rbac_explorer" {
+
+  title         = "Kubernetes RBAC Explorer"
+  documentation = file("./dashboards/rbac/docs/rbac_explorer.md")
+
+  tags = merge(local.rbac_common_tags, {
+    type = "Report"
+  })
+
+  input "cluster_context" {
+    title = "Select a Cluster:"
+    query = query.cluster_input
+    width = 4
+  }
+
+  input "verb" {
+    title = "Select verb(s):"
+    type  = "multiselect"
+    query = query.kubernetes_cluster_verbs
+    width = 4
+    args = {
+      cluster_context = self.input.cluster_context.value
+    }
+  }
+
+  input "resource" {
+    title = "Select resource(s):"
+    type  = "multiselect"
+    query = query.kubernetes_cluster_resources
+    width = 4
+    args = {
+      cluster_context = self.input.cluster_context.value
+    }
+  }
+
+  with "service_accounts_for_rbac" {
+    query = query.service_accounts_for_rbac
+    args = {
+      verb            = self.input.verb.value
+      resource        = self.input.resource.value
+      cluster_context = self.input.cluster_context.value
+    }
+  }
+
+  with "role_bindings_for_rbac" {
+    query = query.role_bindings_for_rbac
+    args = {
+      verb            = self.input.verb.value
+      resource        = self.input.resource.value
+      cluster_context = self.input.cluster_context.value
+    }
+  }
+
+  with "roles_for_rbac" {
+    query = query.roles_for_rbac
+    args = {
+      verb            = self.input.verb.value
+      resource        = self.input.resource.value
+      cluster_context = self.input.cluster_context.value
+    }
+  }
+
+  container {
+    graph {
+      title     = "Kubernetes RBAC Explorer"
+      type      = "graph"
+      direction = "TD"
+      base      = graph.rbac_resource_structure
+      args = {
+        rbac_role_uids            = with.roles_for_rbac.rows[*].uid
+        cluster_role_uids         = with.roles_for_rbac.rows[*].uid
+        role_uids                 = with.roles_for_rbac.rows[*].uid
+        service_account_uids      = with.service_accounts_for_rbac.rows[*].uid
+        role_binding_uids         = with.role_bindings_for_rbac.rows[*].uid
+        cluster_role_binding_uids = with.role_bindings_for_rbac.rows[*].uid
+        rbac_verbs                = self.input.verb.value
+        rbac_resources            = self.input.resource.value
+      }
+
+    }
+  }
+
+  container {
+
+    table {
+      title = "RBAC Analysis"
+      query = query.rbac_rule_analysis
+      args = {
+        verb            = self.input.verb.value
+        resource        = self.input.resource.value
+        cluster_context = self.input.cluster_context.value
+      }
+
+    }
+
+  }
+}
+
+# With queries
+query "rbac_rule_analysis" {
```

Comment:
Should we add sorting to this query?

Reply:
I have added sorting based on `Principal` to the query.